azure-iot-device/samples/async-pnp-samples/sample.py

```diff
@@ -26,158 +28,154 @@
         )
-        self.get_model_definition = Command()
+        self.commands.getModelDefinition = Command()
 
 
 environmental_sensor = EnvironmentalSensor("environmentalSensor")
 device_information = DeviceInformation("deviceInformation")
 model_definition = ModelDefinition("urn_azureiot_ModelDiscovery_ModelDefinition")
-exit_interface = SampleExit("urn_azureiotsdknode_SampleInterface_SampleExit")
+exit_interface = SampleExit("urn_azureiotsdknode_SampleBaseInterface_SampleExit")
 
 
 # Note: this follows the method pattern from the pythonIoTHubDeviceClient object.
-async def environmental_command_listener(interface):
+async def environmental_model_listener(interface):
     while True:
-        command_request = await interface.receive_pnp_command()
+        incoming_message = await interface.receive_pnp_message()
 
-        if command_request.command_name == "blink":
+        if incoming_message.target == interface.commands.blink:
             print("Got the blink command")
 
             command_acknowledge = CommandAcknowledge.create_from_command_request(
-                command_request, 200, "blink response"
+                incoming_message, 200, "blink response"
             )
             try:
                 await interface.send_command_acknowledge(command_acknowledge)
             except Exception:
                 print("responding to the blink command failed")
 
-        elif command_request.command_name == "turnOn":
+        elif incoming_message.target == interface.commands.turnOn:
             # copypasta
             pass
-        elif command_request.command_name == "turnOff":
+        elif incoming_message.target == interface.commands.turnOff:
             # copypasta
             pass
-        elif command_request.command_name == "runDiagnostics":
+        elif incoming_message.target == interface.commands.runDiagnostics:
             print("Got the runDiagnostics command.")
 
             command_acknowledge = CommandAcknowledge.create_from_command_request(
-                command_request, 200, "runDiagnostics response"
+                incoming_message, 200, "runDiagnostics response"
             )
             try:
                 await interface.send_command_acknowledge(command_acknowledge)
             except Exception as e:
                 print("responding to the runDiagnostics command failed: {}".format(e))
             else:
                 command_update = CommandUpdate.create_from_command_request(
-                    command_request, 200, "runDiagnostics update response"
+                    incoming_message, 200, "runDiagnostics update response"
                 )
                 try:
                     await interface.send_command_update(command_update)
                 except Exception as e:
                     print("Got an error on the update: {}".format(e))
+        elif isinstance(incoming_message, PropertyUpdateRequest):
```

Comment:
>lif isinstance(incoming_message, PropertyUpdateRequest): [](start = 9, length = 56)

Note: single listener for commands and property changes.  But, for reusability, each interface needs its own listener.